src/lib/utils.ts

```diff
@@ -9,5 +12,114 @@ function invertPromise<T>(p: Promise<T>): Promise<T> {
 function firstPromiseFulfilled<T>(ps: Promise<T>[]) {
   return invertPromise(Promise.all(ps.map(invertPromise)))
 }
+type GitFS = {
+  readFile(path, options, callback)
+  writeFile(file, data, options, callback)
+  unlink(path, callback)
+  readdir(path, options, callback)
+  mkdir(path, mode, callback)
+  rmdir(path, callback)
+  stat(path, options, callback)
+  lstat(path, options, callback)
+  readlink(path, options, callback)
+  symlink(target, path, type, callback)
+  chmod(path, mode, callback)
+  // For git server
+  exists(path, callback)
+  existsSync(path)
+  mkdirSync(path, options?)
+  createReadStream(path, options)
+}
+
+function efsCallbackWrapper(efs: EncryptedFS): GitFS {
```

Comment:
Wait, this whole function is to enable a callback-version of EFS? That's pretty strange. Didn't VFS have this ability, and then we promisified it?

Reply:
Yeah so this is something I didn't realise until I got to working on the git integration in polykey. Isomorphic git expects the `fs` client to either be a callback or promise based agent and in the case of promises, it expects the methods to be contained in `fs`.`promises` which is something EFS does not do yet. There are also some other discrepancies in so far as `efs` doesn't exactly conform to the `fs` interface. I have been meaning to create an issue for this.

Reply:
I have created an issue here: https://github.com/MatrixAI/js-encryptedfs/issues/32

Reply:
and a PR here: https://github.com/MatrixAI/js-encryptedfs/pull/33
I don't think there is anything stopping us from merging this PR, the callback wrapper was intended to just be a stop-gap until we could just pass in the naked efs object.